src/query/expression/src/schema.rs

```diff
@@ -1528,7 +1534,16 @@ impl From<&DataType> for ArrowDataType {
                 None,
             ),
             DataType::Boolean => ArrowDataType::Boolean,
-            DataType::String => ArrowDataType::LargeBinary,
+            DataType::Binary => ArrowDataType::Extension(
+                ARROW_EXT_TYPE_BINARY.to_string(),
+                Box::new(ArrowDataType::LargeBinary),
+                None,
+            ),
+            DataType::String => ArrowDataType::Extension(
```

Comment:
Can we use arrow2 `LargeUtf8` type for the String type instead of an extension type?

Reply:
Utf8Array::new() will always check utf8. The extension type is used to bypass the check.

Reply:
Oh, Utf8Array::try_new_unchecked seems more appropriate